### DISCUSSION TITLE
"Apply rules" Checkbox on transactions does not save

### DISCUSSION THREAD

**Discussion by Fmstrat** [firefly-iii/firefly-iii#8622] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

Based on https://github.com/firefly-iii/firefly-iii/commit/d31fa743d0d7010a9171ebae674d6dd1e816cb5a and https://github.com/firefly-iii/firefly-iii/issues/6526, unchecking the "Apply rules" when editing a transaction should keep rules from running on that transaction ever again in the future. However, this does not appear to be the case.

### Debug information

Debug information generated at 2024-03-03 14:31:53 for Firefly III version **v6.1.9**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.9 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>957, base <span>#</span>79</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>America/New_York + America/New_York</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>file</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-03-02 00:00:00 (1 day ago)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has rules">:wrench:</span></td></tr><tr><td>Session start</td><td>2024-03-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-03-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0</td></tr></tbody></table>

### Expected behaviour

When unchecking "Apply rules", rules should not be applied to that edited transaction any longer.

### Steps to reproduce

- Create a rule assigning `Category 1` based on description of `Description 1`
- Edit a transaction with `Description 1`
- Change category to `Category 2`
- Uncheck `Apply rules`
- Check `After updating, return here to continue editing` (just to stay on the page)
- Refresh the page
- `Apply rules` is still checked, it should not be

**Comment by Fmstrat** [comment]
A workaround I am using is to set `Internal Reference` to `Locked`, and then adding that as a `not` in my triggers for the rule, but that is pretty cumbersome to have to go back and do across every rule.

**Comment by JC5** [comment]
> When unchecking "Apply rules", rules should not be applied to that edited transaction any longer.

This is not how that button works. It just excludes it the one time.

**Comment by Fmstrat** [comment]
> This is not how that button works. It just excludes it the one time.

In that case, I think https://github.com/firefly-iii/firefly-iii/issues/6526 should be re-opened. This (and their) use case wasn't solved by the commit in question. Thanks!

I should note, even the tag method, where you add a tag of `Locked` and make the first rule stop processing doesn't really work, because if you ever run that rule manually across your account, it doesn't pay attention to the order of the rules, meaning it will change it anyway. You have to put a `Tag not Locked` in every single rule. Not to mention you can't use Rule Groups with that method, either.

**Comment by JC5** [comment]
The button works fine. If you uncheck it, no rules that run when a transaction is updated will fire. That's what the button does. The suggested feature in that issue has not been implemented.

Concerning your second point, this is [documented](https://docs.firefly-iii.org/how-to/firefly-iii/features/rules/#stop-processing)
